allennlp/fairness/bias_direction_wrappers.py

```diff
@@ -0,0 +1,268 @@
+import torch
+from typing import Union
+from os import PathLike
+
+from allennlp.fairness import (
+    PCABiasDirection,
+    PairedPCABiasDirection,
+    TwoMeansBiasDirection,
+    ClassificationNormalBiasDirection,
+)
+from allennlp.fairness.bias_utils import load_word_pairs, load_words
+
+from allennlp.common import Registrable
+from allennlp.data.tokenizers.tokenizer import Tokenizer
+from allennlp.data import Vocabulary
+
+
+class BiasDirectionWrapper(Registrable):
+    """
+    Parent class for bias direction wrappers.
+    """
+
+    def __init__(self):
+        self.direction = None
+        self.noise = None
+
+    def __call__(self, module):
+        raise NotImplementedError
+
+    def train(self, mode: bool = True):
+        """
+
+        # Parameters
+
+        mode : `bool`, optional (default=`True`)
+            Sets `requires_grad` to value of `mode` for bias direction.
+        """
+        self.direction.requires_grad = mode
+
+    def add_noise(self, t: torch.Tensor):
+        """
+
+        # Parameters
+
+        t : `torch.Tensor`
+            Tensor to which to add small amount of Gaussian noise.
+        """
+        return t + self.noise * torch.randn(t.size(), device=t.device)
+
+
+@BiasDirectionWrapper.register("pca")
+class PCABiasDirectionWrapper(BiasDirectionWrapper):
+    """
+
+    # Parameters
+
+    seed_words_file : `Union[PathLike, str]`
+        Path of file containing seed words.
+    tokenizer : `Tokenizer`
+        Tokenizer used to tokenize seed words.
+    direction_vocab : `Vocabulary`, optional (default=`None`)
+        Vocabulary of tokenizer. If `None`, assumes tokenizer is of
+        type `PreTrainedTokenizer` and uses tokenizer's `vocab` attribute.
+    namespace : `str`, optional (default=`"tokens"`)
+        Namespace of direction_vocab to use when tokenizing.
+        Disregarded when direction_vocab is `None`.
+    requires_grad : `bool`, optional (default=`False`)
+        Option to enable gradient calculation for bias direction.
+    noise : `float`, optional (default=`1e-10`)
+        To avoid numerical instability if embeddings are initialized uniformly.
+    """
+
+    def __init__(
+        self,
+        seed_words_file: Union[PathLike, str],
+        tokenizer: Tokenizer,
+        direction_vocab: Vocabulary = None,
```

Comment:
`direction_vocab: Optional[Vocabulary] = None`

Reply:
Have done this for all instances of `Vocabulary = None`